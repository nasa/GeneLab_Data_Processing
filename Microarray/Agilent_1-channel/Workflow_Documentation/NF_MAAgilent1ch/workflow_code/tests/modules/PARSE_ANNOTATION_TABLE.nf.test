nextflow_process {

    name "Test Process PARSE_ANNOTATION_TABLE"
    script "modules/PARSE_ANNOTATION_TABLE.nf"
    process "PARSE_ANNOTATION_TABLE"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = 'https://raw.githubusercontent.com/nasa/GeneLab_Data_Processing/GL_RefAnnotTable_1.0.0/GeneLab_Reference_Annotations/Pipeline_GL-DPPD-7110_Versions/GL-DPPD-7110/GL-DPPD-7110_annotations.csv'
                input[1] = "Homo sapiens"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            assert process.out.annotations_db_url == ["https://figshare.com/ndownloader/files/36597120"]
        }

    }

}
