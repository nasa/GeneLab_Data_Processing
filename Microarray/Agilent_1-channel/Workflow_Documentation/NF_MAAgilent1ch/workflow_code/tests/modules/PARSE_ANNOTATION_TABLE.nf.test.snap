{
    "Should run without failures": {
        "content": [
            {
                "reference_version_and_source": [
                    [
                        "107",
                        "ensembl"
                    ]
                ],
                "annotations_db_url": [
                    "https://figshare.com/ndownloader/files/36597120"
                ]
            }
        ],
        "timestamp": "2023-01-18T00:23:45+0000"
    }
}